src/client.ts

```diff
@@ -1458,10 +1461,17 @@ export class StreamChat<
         event.type === 'notification.channel_deleted') &&
       event.cid
     ) {
-      this.activeChannels[event.cid]?._disconnect();
-      delete this.activeChannels[event.cid];
       client.state.deleteAllChannelReference(event.cid);
+      this.activeChannels[event.cid]?._disconnect();
+
+      postListenerCallbacks.push(() => {
+        if (!event.cid) return;
+
+        delete this.activeChannels[event.cid];
+      });
```

Comment:
```suggestion
      if (event.cid) 
            postListenerCallbacks.push(() => {
              delete this.activeChannels[event.cid];
            });
```

Reply:
We already have the check, this one is just to satisfy typescript!!